src/unix/linux/x11/audio.c

```diff
@@ -54,13 +56,19 @@ MTY_Audio *MTY_AudioCreate(uint32_t sampleRate, uint32_t minBuffer, uint32_t max
 	snd_pcm_hw_params_any(ctx->pcm, params);
 
 	snd_pcm_hw_params_set_access(ctx->pcm, params, SND_PCM_ACCESS_RW_INTERLEAVED);
-	snd_pcm_hw_params_set_format(ctx->pcm, params, SND_PCM_FORMAT_S16);
-	snd_pcm_hw_params_set_channels(ctx->pcm, params, channels);
-	snd_pcm_hw_params_set_rate(ctx->pcm, params, sampleRate, 0);
+	snd_pcm_hw_params_set_format(ctx->pcm, params, format.sampleFormat == MTY_AUDIO_SAMPLE_FORMAT_FLOAT
```

Comment:
are we doing this multiple times, too? a function in cases like this can help minimize the possibility of careless error (though i see no careless error here)

Reply:
This specific highlighted line here doesn't make much sense to be put into a function. It's done exactly once and is very specific to libasound. However, elsewhere I have been using the sample format to determine the size of a sample. I have refactored _that_ into a different function.